signer/migrations/0011__materialized_chains.sql

```diff
@@ -0,0 +1,115 @@
+-- This table holds the canonical chain information for Bitcoin and Stacks
+-- chains keyed by the bitcoin chain tip.
+CREATE TABLE sbtc_signer.canonical_chains (
+    run_id INT,
+    bitcoin_chain_tip BYTEA,
+    bitcoin_block_hash BYTEA,
+    bitcoin_block_height BIGINT,
+    stacks_block_hash BYTEA,
```

Comment:
So apparently we don't have any fk on stacks blocks `bitcoin_anchor` (to bitcoin blocks): given how we process stuff below (joining on the anchor) we should be fine, but if we were to change that nothing would prevent -- data-integrity-wise -- to have null `bitcoin_block_hash`, which could be annoying maybe? So I guess the question is: should we throw some `NOT NULL` in here (for everything but stacks stuff)?

Reply:
Yeah we could do that, bitcoin blocks should _never_ be null here, so better to protect it 👍

Reply:
[34a7709](https://github.com/stacks-network/sbtc/pull/1171/commits/34a77091b9a6f3aeda8f4ee2f2214b1b400078c7)